src/WebformCivicrmPostProcess.php

```diff
@@ -2083,6 +2085,10 @@ private function submitIPNPayment() {
     $params['total_amount'] = round($this->totalContribution, 2);
     // Some processors want this one way, some want it the other
     $params['amount'] = $params['total_amount'];
+    $frequencyUnit = wf_crm_aval($params, 'frequency_unit');
+    if (empty($frequencyUnit)) {
+      unset($params['frequency_unit'], $params['frequency_interval'], $params['is_recur']);
```

Comment:
@jitendrapurohit -> in the LIVE payment section, we use this to determine if it's a recurring contribution:

`if ($numInstallments != 1 && !empty($frequencyInterval)) {`

I think it's best to have the same IF in the IPN section? So use `$params['frequency_interval']` instead of `$frequencyUnit`. Also I'm not even sure if recurring works for IPN Payment Processors.

Reply:
Sounds good to me. Have updated. Thanks @KarinG